### ISSUE TITLE
Deciding Which Architecture to Adopt

### ISSUE THREAD

**Issue by raymondwjang** [Aharoni-Lab/cala#10]
There are two ways to do this that I can think of right now.

1. each module takes a sklearn transformer format, so that they can plug in and out into pipelines. (how the preprocessing package is currently written)
* Pros: 
  - extensible to orchestrators more easily.
  - package structure is more straightforward
* Cons:
  - harder for people to write and contribute
  - not as DRY - need to repeat apply_ufunc in each module.

2. build my own pipeline format (how the motion_correction package is currently written)
* Pros:
  - makes it incredibly easy for people to add in their own modules.
  - DRYer - only need one apply_ufunc
* Cons:
  - the orchestrator looks a bit stinky
  - not sure how well it would integrate into dagster, etc.
  - basically building my own pipeline architecture...?
